src/App.tsx

```diff
@@ -1,24 +1,297 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import {
+  useEffect,
+  FC,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
 import { UserWarning } from './UserWarning';
+import {
+  USER_ID,
+  addTodo,
+  editTodo,
+  getTodos,
+  removeTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { Status } from './types/Status';
 
-const USER_ID = 0;
+const filterTodos = (todos:Todo[], filter:string) => {
+  switch (filter) {
+    case Status.Active:
+      return todos.filter(todo => !todo.completed);
+
+    case Status.Completed:
+      return todos.filter(todo => todo.completed);
+
+    default:
+      return todos;
+  }
+};
+
+export const App: FC = () => {
+  const titleField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>();
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [filter, setFilter] = useState(Status.All);
+  const [unCompletedTodos, setUncompletedTodos] = useState(0);
+  const [allIsCompleted, setAllIsCompleted] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [allCompletedInProcess, setAllCompletedInProcess] = useState(false);
+  const [isDisabled, setIsDisabled] = useState(false);
+
+  useEffect(() => {
+    setErrorMessage('');
+    getTodos()
+      .then(data => setTodos(data))
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+      });
+  }, []);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  }, [errorMessage]);
+
+  useEffect(() => {
+    let allUnCompletedTodos = 0;
+
+    todos.forEach(todo => {
+      allUnCompletedTodos += todo.completed ? 0 : 1;
```

Comment:
```suggestion
      allUnCompletedTodos += +!todo.completed
```

Reply:
Operator '+=' cannot be applied to types 'number' and 'boolean'.ts(2365)

Reply:
Just FYI: you can do this to get a number from boolean:
![image](https://github.com/mate-academy/react_todo-app-add-and-delete/assets/86923089/e791f927-2b3c-46f6-a150-b3fb953e77d7)